File: real-state-backend/internal/core/domain/property_media.go
package domain

import "time"

// PropertyMedia representa un recurso multimedia asociado a una propiedad.
// Se usa para registrar la URL principal de la imagen y mantener el contrato
// del cliente móvil sin introducir una tabla adicional en la base de datos.
type PropertyMedia struct {
	ID         string    `json:"id"`
	PropertyID int64     `json:"property_id"`
	URL        string    `json:"url"`
	Type       string    `json:"type"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
	IsPrimary  bool      `json:"is_primary"`
}

File: real-state-mobile/lib/features/property/domain/entities/property_media.dart
class PropertyMedia {
  const PropertyMedia({
    required this.id,
    required this.propertyId,
    required this.url,
    required this.type,
    this.thumbnailUrl,
    this.width,
    this.height,
    this.sortOrder = 0,
    this.isPrimary = false,
  });

  final String id;
  final String propertyId;
  final String url;
  final String type;
  final String? thumbnailUrl;
  final int? width;
  final int? height;
  final int sortOrder;
  final bool isPrimary;
}
